python/packages/autogen-core/src/autogen_core/components/models/_openai_client.py

```diff
@@ -401,6 +401,10 @@ async def create(
                     **create_args,
                 )
             )
+        elif "response_format" in create_args:
```

Comment:
Instead performing the switching under the hood, can we make it explicit for users by having a `beta` flag in the model client constructor to indicate whether they are using beta features.

Reply:
done